doc/sources/tests.rst

```diff
@@ -0,0 +1,221 @@
+.. Copyright contributors to the oneDAL project
+..
+.. Licensed under the Apache License, Version 2.0 (the "License");
+.. you may not use this file except in compliance with the License.
+.. You may obtain a copy of the License at
+..
+..     http://www.apache.org/licenses/LICENSE-2.0
+..
+.. Unless required by applicable law or agreed to in writing, software
+.. distributed under the License is distributed on an "AS IS" BASIS,
+.. WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. See the License for the specific language governing permissions and
+.. limitations under the License.
+.. include:: substitutions.rst
+
+=============
+Running Tests
+=============
+
+Overview
+--------
+
+The |sklearnex| contains a test suite consisting of a mixture of smoke tests around patching along with unit tests, which are written in a mixture of Python's ``unittest`` (for legacy interfaces) and ``pytest``. However, all of the tests are executed with ``pytest`` as runner. Apart from the tests, code examples are also executed, but are not thoroughly checked for correctness, just for executing without erroring out.
+
+Running test scripts
+--------------------
+
+Requirements
+~~~~~~~~~~~~
+
+As the library is designed with optional components and integrates with external packages that are optional by design, executing the tests involves additional dependencies, some of which are mandatory and some of which are optional.
+
+The mandatory dependencies for tests with locked versions of packages are listed under a file `requirements-test.txt <https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/requirements-test.txt>`__, but it is not strictly necessary to have the exact versions listed there. Those locked requirements can be installed with ``pip`` as follows:
+
+.. code-block:: bash
+
+    pip install -r requirements-test.txt
+
+Some tests will only execute depending on the availability of optional dependencies at runtime. Other optional dependencies that will trigger additional tests can be installed as follows, assuming a Linux* system:
+
+.. code-block:: bash
+
+    pip install \
+        dpctl `# for GPU functionalities` \
+        dpnp `# for array API and GPU functionalities`
+
+    pip install --index-url https://software.repos.intel.com/python/pypi \
+        torch `# for array API`
+
+    pip install --index-url https://software.repos.intel.com/python/pypi \
+        mpi4py impi_rt `# for distributed mode, be sure to install from Intel's index` \
+        && pip install pytest-mpi `# also required, but not from Intel's index`
+
+.. warning:: It might not be possible to install all of the test dependencies simultaneously in the same Python environment. In particular, dependencies ``torch`` and ``dpctl`` / ``dpnp`` are likely to not be installable in a compatible way in the same environment if using pre-built distributions. Try using different Python environments for each set of dependencies to test.
```

Comment:
```suggestion
.. warning:: It might not be possible to install all of the test dependencies simultaneously in the same Python environment. In particular, dependencies of ``torch`` and ``dpctl`` / ``dpnp`` are potentially incompatible if using pre-built distributions. Try using different Python environments for each set of dependencies to test.
```